File: Code C/Exercicio-9/pilha-fila.cpp
#include <stdio.h>
#include <stdlib.h>
int main(){
	/*int op = 10;
	while(op != 0){
		printf("\n Opcoes:");
		printf("\n [0]Sair");
		printf("\n [1] Inserir");
		printf("\n [2] Remover");
		printf("\n [3] Listar \n:_");
		scanf("%d", &op);
		switch(op){
			case 1: //Inserir
			     break;
			case 2: //Remover
			     break;
		    case 3: //Listar
			     break;
		    case 4: //Sair
			     break;
		    default: //Default
		         printf("\nOpcao Invalida\n");
		         break;
	    }
    }*/
    int op = 10, valor = 0, contador = 0, inicio = 0, fim = 0, tamanho = 0;
    int pilha[6];
	while(op != 0){
		printf("\n Opcoes: \n [0]Sair \n [1] Inserir \n [2] Remover \n [3] Listar \n:_");
		scanf("%d", &op);
		switch(op){
			case 1: //Inserir
			    if(tamanho == 6)
			       printf("\n ---Pilha lotada--- \n");
			    else{
			       printf("Digite o valor a ser inserido: ");
			       scanf("%d", &valor);
			       pilha[inicio] = valor;
			       tamanho++;
			       inicio++;
				 }
			     break;
			case 2: //Remover
			    if (tamanho == 0){
			       printf("Lista vazia.");
			    }else{
			       inicio--;
			       pilha[inicio] = 0;
			       tamanho--;
			    }
			    break;
		    case 3: //Listar
		        contador = inicio;
		        while(contador > 0){
			        printf("\n [%d]", pilha[contador-1]);
			        contador--;
				}
			    break;
		    case 0: //Sair
		        printf("\n" 
		                "\n          ------- \n"
				        " ~~> Hasta vista baby"
						"\n          ------- ""\n");
			    break;
		    default: //Default
		        printf("\n Opcao Invalida");
		        break;
	    }
    }
	system("PAUSE");
	return(0);
}

File: Code_Python/Exercicio-14/Pilha-Fila.py
livroBiblioteca = []
escolha = 10
tamanho = 0
while escolha != 0:
    escolha = int(input("Use o programa da seguinte forma/: \n ¬¬>Inserir[1]¬¬>Remover[2]¬¬>Listar[3]¬¬>Sair[0] \n_"))
    if escolha == 0:#Função de Saída
        print("SAIU")
    if escolha == 1:#Função de Inserir
        if tamanho == 7:
            print("Estante Lotada")
        else:
            livroBiblioteca.append(input("Digite o nome do livro:_"))
            tamanho += 1
    if escolha == 2:#Função de Remover
        if tamanho == 0:
            print("Estante vazia, impossível remover.")
        else:
            livroBiblioteca.pop()
            tamanho -= 1
            print(livroBiblioteca)
    if escolha == 3:#Função de Listar
        if tamanho == 0:
            print("Estante vazia, impossível listar.")
        else:
            contador = tamanho - 1
            while contador >= 0:
                print(f"[{livroBiblioteca[contador]}]")
                contador -= 1
            '''if tamanho - 1:
               print(f"[{livroBiblioteca[0:-1]}]")
            else:
               print(livroBiblioteca)'''